File: spring-web-java/src/main/java/spring/web/java/domain/model/entity/RolePermission.java
package spring.web.java.domain.model.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class RolePermission extends Base {

	@Id
	@GeneratedValue
	@Column(name = "role_permission_id")
	private Long id;

	@Id
	@ManyToOne(fetch = FetchType.LAZY)
	@JoinColumn(name = "role_id")
	private Role role;

	@Id
	@ManyToOne(fetch = FetchType.LAZY)
	@JoinColumn(name = "permission_id")
	private Permission permission;

	public static RolePermission create(Permission permission) {
		RolePermission rolePermission = new RolePermission();

		rolePermission.permission = permission;

		return rolePermission;
	}
}

File: spring-web-kotlin/src/main/kotlin/spring/web/kotlin/domain/model/entity/RolePermission.kt
package spring.web.kotlin.domain.model.entity

import jakarta.persistence.*

@Entity
class RolePermission protected constructor(
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "permission_id", foreignKey = ForeignKey(ConstraintMode.NO_CONSTRAINT))
    var permission: Permission,
) {
    @Id
    @GeneratedValue
    @Column(name = "role_permission_id")
    var id: Long? = null
        protected set

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "role_id", foreignKey = ForeignKey(ConstraintMode.NO_CONSTRAINT))
    var role: Role? = null
        protected set

    companion object {
        fun create(permission: Permission) = RolePermission(permission = permission)
    }

    fun associateRole(role: Role) {
        this.role = role
    }
}